rules/REQUEST-922-MULTIPART-ATTACK.conf

```diff
@@ -0,0 +1,115 @@
+# ------------------------------------------------------------------------
+# OWASP ModSecurity Core Rule Set ver.4.0.0-rc1
+# Copyright (c) 2006-2020 Trustwave and contributors. All rights reserved.
+# Copyright (c) 2021-2022 Core Rule Set project. All rights reserved.
+#
+# The OWASP ModSecurity Core Rule Set is distributed under
+# Apache Software License (ASL) version 2
+# Please see the enclosed LICENSE file for full details.
+# ------------------------------------------------------------------------
+
+#
+# -= Paranoia Level 0 (empty) =- (apply unconditionally)
+#
+
+# This file is to address the 3UWMWA6W vulnerability.
+# It requires the latest version of ModSecurity 2.x/3.x or newer, where the variable
+# MULTIPART_PART_HEADERS was introduced.
+
+SecRule TX:DETECTION_PARANOIA_LEVEL "@lt 1" "id:922011,phase:1,pass,nolog,skipAfter:END-REQUEST-922-MULTIPART-ENFORCEMENT"
+SecRule TX:DETECTION_PARANOIA_LEVEL "@lt 1" "id:922012,phase:2,pass,nolog,skipAfter:END-REQUEST-922-MULTIPART-ENFORCEMENT"
+
+#
+# -= Paranoia Level 1 (default) =- (apply only when tx.detection_paranoia_level is sufficiently high: 1 or higher)
+#
+
+# Only allow specific charsets when using "_charset_"
+# Note: this is in phase:2 because these are headers that come in the body
+SecRule &MULTIPART_PART_HEADERS:_charset_ "!@eq 0" \
+    "id:922100,\
+    phase:2,\
+    block,\
+    t:none,t:lowercase,\
+    msg:'Multipart content type charset is not allowed by policy',\
+    logdata:'Matched Data: %{ARGS._charset_}',\
+    tag:'application-multi',\
+    tag:'language-multi',\
+    tag:'platform-multi',\
+    tag:'attack-multipart-header',\
+    tag:'OWASP_CRS',\
+    tag:'capec/1000/255/153',\
+    tag:'paranoia-level/1',\
+    ver:'OWASP_CRS/4.0.0-rc1',\
+    severity:'CRITICAL',\
+    setvar:'tx.inbound_anomaly_score_pl1=+%{tx.critical_anomaly_score}',\
+    chain"
+    SecRule ARGS:_charset_ "!@within %{tx.allowed_request_content_type_charset}" \
+        "t:none,t:lowercase"
+
+
+# Only allow specific charsets on multipart text/plain content-type
+# Note: this is in phase:2 because these are headers that come in the body
+SecRule MULTIPART_PART_HEADERS "@rx ^content-type\s*:\s*text/plain;\s*charset=\"?(.*)\"?" \
+    "id:922110,\
+    phase:2,\
+    block,\
+    capture,\
+    t:none,t:lowercase,\
+    msg:'MIME Multipart Header attack detected',\
+    logdata:'Matched Data: %{TX.1} found within %{MATCHED_VAR_NAME}: %{MATCHED_VAR}',\
+    tag:'application-multi',\
+    tag:'language-multi',\
+    tag:'platform-multi',\
+    tag:'attack-multipart-header',\
+    tag:'OWASP_CRS',\
+    tag:'capec/',\
+    tag:'paranoia-level/1',\
+    ver:'OWASP_CRS/4.0.0-rc1',\
+    severity:'CRITICAL',\
+    setvar:'tx.inbound_anomaly_score_pl1=+%{tx.critical_anomaly_score}',\
+    chain"
+    SecRule TX:1 "!@within %{tx.allowed_request_content_type_charset}" \
```

Comment:
If allowed charsets contains `iso-8859-1` but not `iso-8859-15` and the charset is `iso-8859-15` then this rule will not match, even though it should. You probably need to add boundaries here (also above), e.g., 
```
SecRule TX:1 "!@within /%{tx.allowed_request_content_type_charset}/"
```

Reply:
Maybe we can use something similar to what we do in 920480?

Reply:
Updated.